File: NetFramework/DataFilter/VB/UnboundMode/DropDownFilter.vb
Imports C1.DataCollection
Imports C1.DataFilter
Imports C1.Win.DataFilter

Public Class DropDownFilter
    Inherits ToolStripDropDown

    Public Sub New()
        AutoClose = True
        AutoSize = True
        Margin = Padding.Empty
        Padding = Padding.Empty
        DataFilter = New C1DataFilter With {
            .AutoWidthMode = AutoWidthMode.Grow,
            .ShowClearFilterButtons = True,
            .ShowHeader = False
        }
        Dim host = New ToolStripControlHost(DataFilter) With {
            .Margin = Padding.Empty,
            .Padding = Padding.Empty,
            .AutoSize = True,
            .AutoToolTip = False
        }
        Items.Clear()
        Items.Add(host)
    End Sub

    Public Sub ShowFilter(ByVal propertyName As String, ByVal control As Control, ByVal location As Point)
        Dim isFound = False

        For Each Filter In DataFilter.Filters

            If Filter.PropertyName = propertyName Then
                Filter.Expanded = True
                isFound = True
            Else
                Filter.Expanded = False
            End If
        Next

        If isFound Then
            DataFilter.Height = GetPreferredSize(Size.Empty).Height
            Show(control, location)
        End If
    End Sub

    Public Sub AddFilter(ByVal filter As Filter)
        DataFilter.Filters.Add(filter)
    End Sub

    Public Sub ApplyFilter(ByVal dataSource As ISupportFiltering)
        dataSource.FilterAsync(DataFilter.GetFilterExpression())
    End Sub

    Public ReadOnly Property DataFilter As C1DataFilter

End Class

File: Core/DataFilter/CS/DataFilterExplorer/Samples/Controls/DropDownFilter.cs
using C1.DataCollection;
using C1.DataFilter;
using C1.Win.DataFilter;
using System.Drawing;
using System.Windows.Forms;

namespace DataFilterExplorer.Samples
{
    public class DropDownFilter : ToolStripDropDown
    {
        public DropDownFilter()
        {
            AutoClose = true;
            AutoSize = true;
            Margin = Padding.Empty;
            Padding = Padding.Empty;

            DataFilter = new C1DataFilter
            {
                AutoWidthMode = AutoWidthMode.Grow,
                ShowClearFilterButtons = true,
                ShowHeader = false
            };

            var host = new ToolStripControlHost(DataFilter)
            {
                Margin = Padding.Empty,
                Padding = Padding.Empty,
                AutoSize = true,
                AutoToolTip = false
            };

            Items.Clear();
            Items.Add(host);
        }

        public void ShowFilter(string propertyName, Control control, Point location)
        {
            var isFound = false;
            foreach (var filter in DataFilter.Filters)
            {
                if (filter.PropertyName == propertyName)
                {
                    filter.Expanded = true;
                    isFound = true;
                }
                else
                    filter.Expanded = false;
            }
            if (isFound)
            {
                DataFilter.Height = GetPreferredSize(Size.Empty).Height;
                Show(control, location);
            }
        }

        public void AddFilter(Filter filter) => DataFilter.Filters.Add(filter);

        public void ApplyFilter(ISupportFiltering dataSource) => dataSource.FilterAsync(DataFilter.GetFilterExpression());

        public C1DataFilter DataFilter { get; }

        internal void ApplyTheme(C1.Win.Themes.C1Theme theme) => C1.Win.Themes.C1ThemeController.ApplyThemeToControlTree(DataFilter, theme);
    }
}
